dipy/align/bundlemin.pyx

```diff
@@ -242,6 +242,99 @@ def _bundle_minimum_distance(double [:, ::1] stat,
     return dist
 
 
+
+def _bundle_minimum_distance_asymmetric(double [:, ::1] static,
+                                        double [:, ::1] moving,
+                                        cnp.npy_intp static_size,
+                                        cnp.npy_intp moving_size,
+                                        cnp.npy_intp rows):
+    """ MDF-based pairwise distance optimization function
+
+    We minimize the distance between moving streamlines of the same number of
+    points as they align with the static streamlines.
+
+    Parameters
+    -----------
+    static : array
+        Static streamlines
+    moving : array
+        Moving streamlines
+    static_size : int
+        Number of static streamlines
+    moving_size : int
+        Number of moving streamlines
+    rows : int
+        Number of points per streamline
+
+    Returns
+    -------
+    cost : double
+
+    Notes
+    -----
+    The difference with ``_bundle_minimum_distance`` is that we sum the
+    minimum values only for the static. Therefore, this is asymetric.
+    """
+
+    cdef:
+        cnp.npy_intp i=0, j=0
+        double sum_i=0, sum_j=0, tmp=0
+        double inf = np.finfo('f8').max
+        double dist=0
+        double * min_j
+        # double * min_i
+        openmp.omp_lock_t lock
+
+    with nogil:
+
+        if have_openmp:
+            openmp.omp_init_lock(&lock)
+
+        min_j = <double *> malloc(static_size * sizeof(double))
+        # min_i = <double *> malloc(moving_size * sizeof(double))
```

Comment:
Because you only keep the min of the first loop, you don't need an array. A simple `min` and `sum` var will do.